### PR TITLE
User/mglyngberg/sft 119

### PR DESCRIPTION
**Pull request by mglyngberg**
Fixes SFT-119 Add new aux bms telemetry protocol fields to hermes

Rename Contactor Error -> HighVoltageEnableState, 
Rename HighVoltageEnable -> AllowDischarge

Add fields for AuxContactor and AuxTrip

The spreadsheet should be updated once this is merged:
https://docs.google.com/spreadsheets/d/1soVLjeD9Sl7z7Z6cYMyn1fmn-cG7tx_pfFDsvgkCqMU/edit#gid=782574835

### CODE REVIEW

**Comment by mglyngberg** [comment]
Some of the terminology doesn't line up between the serial and internet protocol in the spreadsheet eg.  ChargeContactorError (serial) vs ContactorErrorCharge (internet). They should probably be consistent, but I'm not sure which terminology we prefer. In this pr all of the json keys match up with the serial terminology, so it is out of line with the internet protocol laid out in the spreadsheet.

**Comment by bill-luu** [comment]
Oops, it should have been `ChargeContactorError`, I'll give it another pass to make sure it is consistent

**Review by ThomasVy — APPROVED** [review]

File: src/CommunicationLayer/MessagingFramework/MessageDefines.cpp [hunk 1]
```diff
@@ -3,7 +3,7 @@
 namespace
 {
 // Message lengths do not include metadata
-    const int AUX_BMS_DATA_LENGTH = 8;
+    const int AUX_BMS_DATA_LENGTH = 23;
```
Comment (bill-luu) [inline src/CommunicationLayer/MessagingFramework/MessageDefines.cpp]: Shouldn't this be 17, not 23? Given the last element at index 16 is a uchar

File: src/DataLayer/AuxBmsData/AuxBmsData.cpp [hunk 2]
```diff
@@ -47,14 +64,89 @@ bool AuxBmsData::getAllowCharge() const
...
+}
+
+bool AuxBmsData::getAllowDischarge() const
+{
+    return allowDischarge_;
+}
+
+bool AuxBmsData::getOrionCANReceivedRecently() const
+{
+    return auxMaskedBit(ORION_CAN_RECEIVED_RECENTLY_MASK);
```
Comment (bill-luu) [inline src/DataLayer/AuxBmsData/AuxBmsData.cpp]: You have `auxMaskedBit` called with `ORION_CAN_RECEIVED_RECENTLY_MASK` and `CHARGE_TRIP_DUE_TO_HIGH_CELL_VOLTAGE_MASK`, which are both `0x01`. Did you mean to create a `auxContactorMaskedBit` function too?

**Review by bill-luu — CHANGES_REQUESTED** [review]

**Review by bill-luu — APPROVED** [review]

**Review by bill-luu — APPROVED** [review]

**Review by bill-luu — CHANGES_REQUESTED** [review]
`auxContactorDebug`->`auxContactorDebugInfo`, to match up with the spreadsheet <3

**Review by bill-luu — APPROVED** [review]

**Pull request merged**